src/angular-breadcrumb.js

```diff
@@ -255,17 +261,22 @@ function BreadcrumbLastDirective($interpolate, $breadcrumb, $rootScope) {
                     var renderLabel = function() {
                         deregisterWatchers(labelWatchers);
                         labelWatchers = [];
-                        
+
                         var viewScope = $breadcrumb.$getLastViewScope();
                         var lastStep = $breadcrumb.getLastStep();
                         if(lastStep) {
                             scope.ncyBreadcrumbLink = lastStep.ncyBreadcrumbLink;
                             if (lastStep.ncyBreadcrumb && lastStep.ncyBreadcrumb.label) {
-                                var parseLabel = $interpolate(lastStep.ncyBreadcrumb.label);
-                                scope.ncyBreadcrumbLabel = parseLabel(viewScope);
-                                // Watcher for further viewScope updates
-                                // Tricky last arg: the last step is the entire scope of the directive !
-                                registerWatchers(labelWatchers, parseLabel, viewScope, scope);
+                                var type = Object.prototype.toString.call(lastStep.ncyBreadcrumb.label);
+                                if (type === '[object Function]' ||
+                                    type === '[object Array]') {
+                                    lastStep.ncyBreadcrumbLabel = $injector.invoke(lastStep.ncyBreadcrumb.label);
+                                } else {
+                                    var parseLabel = $interpolate(lastStep.ncyBreadcrumb.label);
+                                    lastStep.ncyBreadcrumbLabel = parseLabel(viewScope);
```

Comment:
For this directive, we directly assign the variables to the scope of the directive.

```javascript
scope.ncyBreadcrumbLabel = parseLabel(viewScope);
```

The same above (L273).

It's the cause of the test failure :smile: